File: app/src/main/java/com/laurelkeys/userinformationtestingrobots/userinformation/model/UserInformation.java
package com.laurelkeys.userinformationtestingrobots.userinformation.model;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.laurelkeys.userinformationtestingrobots.userinformation.UserInformationMVP;
import com.laurelkeys.userinformationtestingrobots.utils.Utils;

import java.io.Serializable;

/**
 * Created by Tiago in July 2016.
 */
public class UserInformation implements UserInformationMVP.Model, Serializable {

    private static final String TAG = UserInformation.class.getSimpleName();

    public String firstName;
    public String lastName;
    public String phoneNumber;
    public String email;
    public String age;

    public UserInformation(@NonNull String firstName, @NonNull String lastName,
                           @Nullable String phoneNumber, @NonNull String email, @NonNull String age) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.phoneNumber = phoneNumber;
        this.email = email;
        this.age = age;
    }


    @Override
    public boolean isValidName(@NonNull String name) {
        return name.matches(Utils.Regex.NAME);
    }

    @Override
    public boolean isValidPhoneNumber(@Nullable String phoneNumber) {
        return phoneNumber != null && phoneNumber.matches(Utils.Regex.PHONE_NUMBER);
    }

    @Override
    public boolean isValidEmail(@NonNull String email) {
        return email.matches(Utils.Regex.EMAIL);
    }

    @Override
    public boolean isValidAge(@NonNull String age) {
        return age.matches(Utils.Regex.AGE);
    }


    @Override
    public String toString() {
        return "[firstName: " + firstName
                + ", lastName: " + lastName
                + ", phoneNumber: " + (phoneNumber == null ? "null" : phoneNumber)
                + ", email: " + email
                + ", age: " + age
                + "]";
    }
}

File: app/src/main/kotlin/com/laurelkeys/userinformationtestingrobots/userinformation/model/UserInformation.kt
package com.laurelkeys.userinformationtestingrobots.userinformation.model

import com.laurelkeys.userinformationtestingrobots.userinformation.UserInformationContract
import com.laurelkeys.userinformationtestingrobots.userinformation.model.UserInformation.Regex.*
import java.io.Serializable

/**
 * Created by Tiago in July 2016.
 */
data class UserInformation(var firstName: String, var lastName: String,
                           var email: String, var age: String) : UserInformationContract.Model, Serializable {

    companion object { private val TAG = UserInformation::class.simpleName ?: "UserInformation" }

    enum class Regex(val regex: kotlin.text.Regex) {
        NAME("^[\\p{L} .'-]+$".toRegex()),
        EMAIL("^[A-Za-z0-9_.]+@[A-Za-z0-9_.]+\\.[A-Za-z]{2,3}$".toRegex()),
        AGE("^[0-9]{1,2}$".toRegex()),
    }

    override fun isValidName(name: String): Boolean {
        return name.matches(NAME.regex)
    }

    override fun isValidEmail(email: String): Boolean {
        return email.matches(EMAIL.regex)
    }

    override fun isValidAge(age: String): Boolean {
        return age.matches(AGE.regex)
    }
}
